hail/python/hail/experimental/vcf_combiner.py

```diff
@@ -53,21 +53,39 @@ def transform_one(mt: MatrixTable) -> MatrixTable:
 
     return mt
 
+def combine(ts):
+    def merge_alleles(alleles):
+        return hl.rbind(
+            alleles.map(lambda a: hl.or_else(a[0], ''))
+                   .fold(lambda s, t: hl.cond(hl.len(s) > hl.len(t), s, t), ''),
+            lambda ref:
+            hl.rbind(
+                alleles.map(
+                    lambda al: hl.rbind(
+                        al[0],
+                        lambda r:
+                        hl.array([ref]).extend(
+                            al[1:].map(
+                                lambda a:
+                                hl.switch(hl.allele_type(r, a))
+                                  .when('SNP', a + ref[hl.len(a):])
+                                  .when('Insertion', a + ref[hl.len(r):])
+                                  .when('Deletion', a + ref[hl.len(r):])
+                                  .default(a))))),
```

Comment:
I realize that this logic is preserved from before, but I feel like this may pad some cases incorrectly. Why can't this just be:
```
hl.switch(hl.allele_type(r, a))
  .when('Star', a)
  .when('Symbolic', a)
  .when('Unknown', a)
  .default(a + ref[hl.len(r):])
```
? (The cases I'm thinking of specifically are 'MNP' and 'Complex', but I feel like there's fewer cases to check in the other direction.)